cpp/src/arrow/sparse_tensor.h

```diff
@@ -29,42 +29,58 @@ namespace arrow {
 // ----------------------------------------------------------------------
 // SparseIndex class
 
-/// \brief EXPERIMENTAL: Sparse tensor format enumeration
 struct SparseTensorFormat {
+  /// EXPERIMENTAL: The index format type of SparseTensor
   enum type { COO, CSR };
 };
 
-/// \brief EXPERIMENTAL: The base class for representing index of non-zero
-/// values in sparse tensor
+/// \brief EXPERIMENTAL: The base class for the index of a sparse tensor
+///
+/// SparseIndex are used to represent where is non-zero values in a
+/// SparseTensor.
+///
+/// There are several ways of the index representation.  The format_id is used
```

Comment:
I think it'd be useful to note specifically that there is a single derived class for each possible value of format_id (`SparseTensorFormat::COO -> SparseCOOIndex` and `SparseTensorFormat::CSR -> SparseCSRIndex`)

Reply:
I tried to describe that you mentioned.
Could you please review it again?

Reply:
Thanks!
Also this sentence isn't quite grammatically correct. Could you use `-ways of the index representation+ways to represent this`?

Reply:
Thank you for your feedback.  I fixed the incorrect representation.